lib/kitchen/provisioner/ansible_push.rb

```diff
@@ -88,7 +89,11 @@ def conf
 
       def machine_name
         return @machine_name if defined? @machine_name
-        @machine_name = instance.name.gsub(/[<>]/, '').split("-").drop(1).join("-")
+        unless config[:use_instance_name]
+          @machine_name = instance.platform.name.gsub(/[<>]/, '')
```

Comment:
From our previous discussion on the issue, I thought this was on the spirit of your original intention. But you're right, if it is considered a breaking change we can leave the previous implementation and push this on 0.4. Let me know and I'll update the PR.